core/math/convex_hull.cpp

```diff
@@ -2268,10 +2268,20 @@ Error ConvexHullComputer::convex_hull(const Vector3 *p_points, int32_t p_point_c
 
 	r_mesh.vertices = ch.vertices;
 
-	r_mesh.edges.resize(ch.edges.size());
+	// Copy the edges over. There's two "half-edges" for every edge, so we pick only one of them.
+	r_mesh.edges.resize(ch.edges.size() / 2);
+	uint32_t edges_copied = 0;
 	for (uint32_t i = 0; i < ch.edges.size(); i++) {
-		r_mesh.edges.write[i].a = (&ch.edges[i])->get_source_vertex();
-		r_mesh.edges.write[i].b = (&ch.edges[i])->get_target_vertex();
+		uint32_t a = (&ch.edges[i])->get_source_vertex();
+		uint32_t b = (&ch.edges[i])->get_target_vertex();
+		if (a < b) { // Copy only the "canonical" edge. For the reverse edge, this will be false.
+			r_mesh.edges.write[edges_copied].a = a;
```

Comment:
To be perfectly safe, it would be nice to add an error in case `edge_copied` didn't fit the vector so it wouldn't cause memory corruption if it happened.